tests/conftest.py

```diff
@@ -187,12 +189,14 @@ async def guard_func(*args, **kwargs):
 
 
 @pytest.fixture(autouse=True)
-def verify_cleanup():
+def verify_cleanup(event_loop: asyncio.AbstractEventLoop):
     """Verify that the test has cleaned up resources correctly."""
     threads_before = frozenset(threading.enumerate())
-
+    tasks_before = asyncio.all_tasks(event_loop)
     yield
 
+    event_loop.run_until_complete(event_loop.shutdown_default_executor())
+
```

Comment:
If we could ignore the executor threads easily in the below threads check, we dont need to stop these here.